infrastructure/lib/stacks/gitHubAutomationApp.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+import {Aspects, Duration, Stack, Tag, Tags} from 'aws-cdk-lib';
+import {
+    AutoScalingGroup,
+    BlockDeviceVolume,
+    CfnLaunchConfiguration,
+    HealthCheck,
+    UpdatePolicy
+} from 'aws-cdk-lib/aws-autoscaling';
+import {
+    InstanceClass,
+    InstanceSize,
+    InstanceType,
+    MachineImage,
+    SubnetType,
+    Vpc
+} from 'aws-cdk-lib/aws-ec2';
+import {
+    ArnPrincipal,
+    CompositePrincipal,
+    Effect,
+    ManagedPolicy,
+    PolicyStatement,
+    Role,
+    ServicePrincipal
+} from 'aws-cdk-lib/aws-iam';
+import { Construct } from 'constructs';
+import {OpenSearchMetricsSecretsStack} from "./secrets";
+import {Secret} from "aws-cdk-lib/aws-secretsmanager";
+
+
+export interface GitHubAppProps {
+    readonly vpc: Vpc;
+    readonly region: string;
+    readonly account: string;
+    readonly ami?: string
+    readonly secret: Secret;
+}
+
+
+export class GitHubAutomationApp extends Stack {
+
+    readonly asg: AutoScalingGroup;
+    readonly githubAppRole: Role;
+
+    constructor(scope: Construct, id: string, props: GitHubAppProps) {
+        super(scope, id);
+
+        const instanceRole = this.createInstanceRole(props.secret.secretArn, props.account);
+        this.githubAppRole = instanceRole;
+
+        this.asg = new AutoScalingGroup(this, 'OpenSearchMetrics-GitHubAutomationApp-Asg', {
+            instanceType: InstanceType.of(InstanceClass.M5, InstanceSize.LARGE),
+            blockDevices: [{ deviceName: '/dev/xvda', volume: BlockDeviceVolume.ebs(10) }],
```

Comment:
10G block enough?

Reply:
I have changed to 20gb, since its just the docker logs we can start with it, later we can update to `--log-driver=awslogs `.